drivers/usb/serial/xr_usb_serial_hal.c

```diff
@@ -500,6 +500,40 @@ int xr_usb_serial_set_flow_mode(struct xr_usb_serial *xr_usb_serial, struct tty_
 
 	xr_usb_serial_set_reg(xr_usb_serial, xr_usb_serial->reg_map.uart_flow_addr, flow);
 	xr_usb_serial_set_reg(xr_usb_serial, xr_usb_serial->reg_map.uart_gpio_mode_addr, gpio_mode);
+
+	return 0;
+}
+
+int xr_usb_serial_set_rs485_mode(struct xr_usb_serial *xr_usb_serial, bool enabled, bool hd)
+{
+	unsigned int flow;
+	unsigned int gpio_mode;
+
+	if  (enabled) {
+		dev_info(&xr_usb_serial->control->dev, "xr_usb_serial_set_rs485_mode: RS485 mode on\n");
+
+		if (hd) {
+			dev_info(&xr_usb_serial->control->dev, "xr_usb_serial_set_rs485_mode: RS485 Half-duplex mode\n");
+			flow = UART_FLOW_MODE_HALF_DUPLEX;
+		}
+		else
+		{
+			dev_info(&xr_usb_serial->control->dev, "xr_usb_serial_set_rs485_mode: RS485 Full-duplex mode\n");
```

Comment:
If you are enabling the Full-duplex mode, the description in the Kconfig file must be updated as well...

Reply:
@rene sorry, but I do not quite understand it. are you proposing to remove "	tristate "USB MaxLinear/Exar USB to Serial driver in RS485 Half-duplex mode" Half-duplex part?

Reply:
or add `Half/Full-duplex` mode.... from the description it clear says it only supports Half-duplex mode, but you have added support for Full-duplex mode as well, right?